jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java

```diff
@@ -260,6 +267,8 @@ MethodHandle cloneWithNewType(MethodType newType) {
 		enforceArityLimit(kind, this.type);
 		/* Must be called even laster as it uses the method type */
 		this.thunks = computeThunks(thunkArg);
+		/* Touch thunks.invokeExactThunk so that its constant pool entry is resolved by the time it is used by the JIT */
+		long i = thunks.invokeExactThunk;
```

Comment:
There is a static block in this class:
```
	static {
		// TODO:JSR292: Get rid of these if possible
		ComputedCalls.load();
		ThunkKey.load();
		ThunkTuple.load();
		ILGenMacros.load();
	}
```
that forces the ThunkTuple to be loaded.  Can this forced cp resolve be moved there?  It seems like a waste to touch it in every constructor.

Reply:
Both `thunks` and `invokeExactThunk` are instance field, they cannot be accessed in a static block, right?

Reply:
Right, these are instance fields so we could create an instance and touch the fields.  

I'm OK with leaving this in the ctor if the jit is expected to remove the unnecessary load when it compiles the ctor.

Reply:
This is due to the cold block marking optimizations?  Do we have any other control mechanisms for cold block marking - whether to ignore specific unresolved fields or skip certain methods?

Reply:
Special hardcoded fields/methods could be added to cold block marking, but this isn't done at present. Simply forcing the field to resolve in the constant pool is all that is needed. The dead local should be cleaned up by JIT compilation and so the load should cease to happen at that point. Having the field resolve rather than adding a special case to the JIT seems nicer to me, but happy to hear alternative arguments.

Reply:
I'm OK with leaving this field load here then based on Andrew's comments

Reply:
One additional thought - should we construct an instance of this handle in the static init for the class? we could touch the field there then if we didn't want it in the constructor. Thoughts @DanHeidinga ?

Reply:
After thinking about this, the field load in the ctor is probably the right solution as this will only be a cost to the interpreter (and a small one at that).  Creating an MH instance in its ctor may load other classes early in startup which is likely to be a higher cost (though probably unnoticeable)